[doc] Add doc on SHAPEs and INTRUDERs. (#2815)
paparazzi/paparazzi: doc/sphinx/source/user_guide/gcs_graphics.rst

.. user_guide flight_plan

================
GCS map graphics
================

GCS map can be enriched with intruders and shapes.

Intruders
---------

> Intruders represents vehicules that are not managed by paparazzi. They are updated on the `INTRUDER` message.

Reviewer: ```suggestion
Intruders represents vehicules that are not managed by paparazzi. They are updated with the `INTRUDER` message.
```

.. image:: intruder.png

The **INTRUDER** message is defined as follow:

.. code-block:: xml

    <message name="INTRUDER" id="37">
      <field name="id"     type="string"/>
      <field name="name"   type="string"/>
      <field name="lat"    type="int32"  unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"/>
      <field name="lon"    type="int32"  unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"/>
      <field name="alt"    type="int32"  unit="mm" alt_unit="m">altitude above WGS84 reference ellipsoid</field>
      <field name="course" type="float"  unit="deg"/>
      <field name="speed"  type="float" unit="m/s"/>
      <field name="climb"  type="float"  unit="m/s"/>
      <field name="itow"   type="uint32" unit="ms"/>
    </message>

Latitudes and longitudes are integers in 1e7deg: ``43.123456`` become ``431234560``.

.. note::

    You can use the **OpenSky-Network Intruders** tool (in the paparazzi center: Tools->OpenSky-Network Intruders) to view ADS-B aircrafts near your ACs.
    
    The **id** field will be the aircraft's hexadecimal ICAO 24 bits address, and the **name** field its callsign.

Shapes
------

> Shapes can be draw in the GCS by sending the **SHAPE** message defined below.

Reviewer: ```suggestion
Shapes can be drawn in the GCS by sending the **SHAPE** message defined below.
```

        </description>
      <field name="id"         type="uint8"/>
      <field name="linecolor"  type="string"/>
      <field name="fillcolor"  type="string"/>
      <field name="opacity"    type="uint8" values="Transparent|Light|Medium|Opaque"/>
      <field name="shape"      type="uint8" values="Circle|Polygon|Line"/>
      <field name="status"     type="uint8"  values="create|delete"/>
      <field name="latarr"     type="int32[]"  unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"/>
      <field name="lonarr"     type="int32[]"  unit="1e7deg" alt_unit="deg" alt_unit_coef="0.0000001"/>
      <field name="radius"     type="float" unit="m"/>
      <field name="text"       type="string"/>
    </message>

The embedded documentation is pretty explicit here.

How is this different from sectors, and which one should I choose?
__________________________________________________________________

Sectors are defined in the flight plan with waypoints, and are embedded in the aircraft. The aircraft can react to it even if telemetry is down.
It is advised to keep the number of waypoints low, so this limits the sector "resolution".

> Shapes, on the other hand, are simply drawn in the GCS. This mean that the aircraft has no knowledge of their existance, so it can't react to them by itself.

Reviewer: ```suggestion
Shapes, on the other hand, are simply drawn in the GCS. It means that the aircraft has no knowledge of their existence, so it can't react to them by itself.
```
